src/components/inputModal/index.tsx

```diff
@@ -1,62 +1,93 @@
+/* eslint-disable react-hooks/exhaustive-deps */
 import { Modal, Form } from "react-bootstrap";
-import React, { useEffect } from "react";
+import React, { useEffect, useMemo } from "react";
 import { useForm, Controller } from "react-hook-form";
 import { yupResolver } from "@hookform/resolvers/yup";
 import FormButtonGroup from "components/inputModal/FormAction";
 import { schema } from "utils/inputFormSchema";
 import { MODAL_TYPE } from "utils/enums";
-import InputTextField from "components/inputModal/InputTextField";
-import InputSelectField from "components/inputModal/InputSelect";
-import {grades, subjects} from "utils/index";
+import InputTextField from "components/inputModal/inputTextField";
+import InputSelectField from "components/inputModal/inputSelect";
+import { grades, subjects } from "utils/index";
+import { IStudentRaw } from "state/ducks/students/types";
+import { ActionType } from "typesafe-actions";
+import { addStudent, updateStudent } from "state/ducks/students/actions";
 interface IModalProps {
   visible: boolean;
-  setVisible: Function;
-  // isEdit: boolean;
+  setVisible: (x: boolean) => void;
+  addStudent: (payload: IStudentRaw) => ActionType<typeof addStudent>;
+  updateStudent: (payload: IStudentRaw) => ActionType<typeof updateStudent>;
+  studentData: IStudentRaw;
+  setStudentData: (std: IStudentRaw) => void;
 }
 
 const StudentInputModal: React.FC<IModalProps> = (props) => {
-  const initState = {
+  const blankForm = {
     name: "",
+    marks: 0,
     subject: "",
-    marks: "",
-    grades: "",
+    grade: "",
   };
 
-  // const modalType = props?.isEdit ? MODAL_TYPE.EDIT : MODAL_TYPE.ADD;
-  const modalType = MODAL_TYPE.ADD;
+  const modalType = props?.studentData?._id ? MODAL_TYPE.EDIT : MODAL_TYPE.ADD;
 
   const {
-    register,
     handleSubmit,
     reset,
     control,
-    formState: { errors, isSubmitSuccessful },
-  } = useForm({
+    formState: { errors },
+  } = useForm<IStudentRaw>({
     mode: "onTouched",
     reValidateMode: "onChange",
     resolver: yupResolver(schema),
-    defaultValues: initState,
+    defaultValues: useMemo(() => {
+      return props.studentData || blankForm;
+    }, [props.studentData]),
   });
 
   useEffect(() => {
-    if (isSubmitSuccessful) {
-      reset();
-      console.log("Reset form!");
+    reset(props.studentData);
+  }, [props.studentData]);
```

Comment:
Isn’t this repetitive? You’re setting the default values and resetting them again.

Reply:
I've reviewed several links online, and they all recommend using defaultValues together with resetForm()